### DISCUSSION TITLE
Error: The fulfiller does not have the balances needed to fulfill

### DISCUSSION THREAD

**Discussion by BravoNatalie** [ProjectOpenSea/seaport-js#122] [Q&A]
I'm trying to use seaport to make it possible to create purchase orders, but I'm confused about the parameters I have to pass and I believe this might be leading me to this error.

Is the fulfiller equal to the offerer?

The buyer account has enough ETH to buy the item, what could be the reason for the `Error: The fulfiller does not have the balances needed to fulfill` error?

```
const { executeAllActions } = await seaport.createOrder(
      {
        offer: [
          {
            amount: ethers.utils
              .parseEther(asset.sellOrder.currentPrice)
              .toString(),
          },
        ],
        consideration: [
          {
            itemType: ItemType.ERC721,
            token: "0x13b9ba7c61c7774089fbfce6cffc22021fb14110",
            recipient: buyerAddress,
            identifier: asset.tokenId,
          },
        ],
      },
      buyerAddress
    );

    const order = await executeAllActions();

    const { executeAllActions: executeAllFulfillActions } =
      await seaport.fulfillOrder({
        order,
        accountAddress: buyerAddress,
      });

    const transaction = await executeAllFulfillActions();
   ```

**Comment by Maumasi** [comment]
I'm running into the same issue. I have essentially the same code example.

**Comment by 0xAryanShah** [comment]
Any luck guys?
if anybody seeing this can post a working fulfillorder , will help a lot

**Comment by ryanio** [comment]
> Is the fulfiller equal to the offerer?
No the fulfiller is the "buyer", the one executing the order on chain.

So in createOrder you should pass the sellerAddress as both the recipient of the consideration item and the accountAddress.

Then in fulfillOrder you should pass the buyer address.

In your order in the opening post, the offer should be the NFT (the offerer is offering the NFT for sale) and the consideration should be the amount the person is willing to sell it for, along with the fees to the right parties (creator fees, marketplace fees). Would recommend using [opensea-js](https://github.com/ProjectOpenSea/opensea-js) if you are looking to specifically list on opensea as it will take care of this for you.

**Reply by 0xAryanShah** [reply]
@ryanio  thanks for the help,
so i was able to create a order (basically i was able to create a Ask for my nft or list nft as you say)
then validated it, so it went on chain.

now with other account i want to fill that ask and buy the NFT,
for that i am using fulfillOrder,
in that approach i am using is
1) create order with offerer and receipient as the seller.
2) swap the intial consideration wth offer and offer with consideration,
3) then execute and await the order, then use that order to call fulfillOrder.
4) and account address for fullfillOrder is buyerAddress

posting code for filling for reference,

i am getting error and its not buying nft, see if you can help

```
const { data: signer } = useSigner();
  const main = async () => {
    // @ts-ignore
    const seaport = new Seaport(signer);
    console.log(seaport);
    const offerer = '0x31246AFAb38328dE8181068765F749E614993767'; // address of the NFT owner

    const conduitKey = '0x0000007b02230091a7ed01230072f7006a004d60a8d4e71d599b8104250f0000';

    const { executeAllActions } = await seaport.createOrder(
      {
        zone: '0x0000000000000000000000000000000000000000',
        zoneHash: '0x0000000000000000000000000000000000000000000000000000000000000000',
        startTime: Math.floor(new Date().getTime() / 1000).toString(),
        endTime: Math.floor(new Date().getTime() / 1000 + 6000 * 6000).toString(),
        offer: [
          {

            amount: ethers.utils.parseEther('3').toString(),
          },
        ],
        consideration: [
          {
            itemType: 3,
            token: '0x2f3BCd2dd8A6491eA893eeE88732DFea126D7E42',
            identifier: '1',
            amount: '1',
            recipient: offerer,
          },
        ],
        conduitKey,
      },
      offerer,
    );

    const order = await executeAllActions();
    // eslint-disable-next-line no-console

    // this will be call by signer 1 to fulfill the order
    const { executeAllActions: executeAllFulfillActions } = await seaport.fulfillOrder({
      order, // providing the order which we are fulfilling
      accountAddress: currentUserAddress, // the address through which we will fulfill
    });

    const transaction = await executeAllFulfillActions(); // getting the transaction details

    console.log('transaction success');
    console.log(transaction);

    // const validated = seaport.validate([order]);
    // const transaction = await validated.transact();
    // // eslint-disable-next-line no-console
    // console.info('Validated: ', await transaction.wait());
  };
```

what is wrong with this??
please help

**Comment by tenthirtyone** [comment]
This happens when I use a stale order, one that is no longer valid.
